jena-arq/Grammar/master.jj

```diff
@@ -100,6 +100,38 @@ import org.apache.jena.sparql.core.Quad ;
 public class CLASS extends PARSERBASE
 {
     boolean allowAggregatesInExpressions = false ;
+
+    public static void main(String args[]) {
+        while (true) {
```

Comment:
New `CLASS` don't appear very often so this could be in a java source file for each language

In fact, it only needs to work for language ARQ which is a superset of SPARQL 1.1 

Also - have you seen the command `arq.qparse`? It reads in a query and prints it out (and performs internal checks on `.equals`, `.hashcode`, output sameas input and the algebra generated.